robo-components/DeploymentTrait.php

```diff
@@ -254,8 +251,13 @@ public function deployPantheon(string $branch_name = 'master', ?string $commit_m
     $this->_exec("cp web/sites/default/settings.pantheon.php $pantheon_directory/web/sites/default/settings.php");
 
     // Prevent attackers to reach these standalone scripts.
-    $this->_exec("rm -rf web/core/install.php");
-    $this->_exec("rm -rf web/core/update.php");
+    $this->_exec("rm -f $pantheon_directory/web/core/install.php");
+    $this->_exec("rm -f $pantheon_directory/web/core/update.php");
+
+    // Remove the dev dependencies before pushing up to Pantheon.
+    $this->taskExec("rm -rf $pantheon_directory/vendor")->run();
+    $this->taskExec("(cd $pantheon_directory && composer install --no-dev)")->run();
+    $this->taskExec("(cd $pantheon_directory && composer dump-autoload)")->run();
```

Comment:
`_exec()` is a shortcut for `taskExec()->run()`. Also can we chain dump-autoload?

```suggestion
    $this->_exec("rm -rf $pantheon_directory/vendor");
    $this->_exec("(cd $pantheon_directory && composer install --no-dev && composer dump-autoload)");
```